numpy/core/tests/test_multiarray.py

```diff
@@ -403,6 +403,13 @@ def test_fill_struct_array(self):
         assert_array_equal(x['a'], [3.5, 3.5])
         assert_array_equal(x['b'], [-2, -2])
 
+    def test_fill_readonly(self):
+        # gh-22922
+        a = np.zeros(11)
+        a.setflags(write=False)
+        with pytest.raises(ValueError):
```

Comment:
```suggestion
        with pytest.raises(ValueError, match=".*read-only"):
```
Can't hurt to be even more specific, didn't think of it.